networkit/cpp/centrality/test/CentralityGTest.cpp

```diff
@@ -1293,6 +1294,36 @@ TEST_P(CentralityGTest, testTopCloseness) {
     }
 }
 
+TEST_F(CentralityGTest, testTopClosenessWithNodeList) {
+    METISGraphReader reader;
+    Graph G = reader.read("input/lesmis.graph");
+    constexpr count k = 10;
+    const std::vector<node> nodeList{0, 1, 2, 3, 4, 5, 11, 26, 48, 64};
+
+    // We expect complete TopCloseness to not contain nodes 0-5, while
+    // restricted should. The first element for both complete and
+    // restricted TopCC should be 11. {26, 48, 64} should also be present
+    // in both results (but position might differ).
+    for (auto firstHeu : {true, false}) {
+        for (auto secHeu : {true, false}) {
+            TopCloseness topC(G, k, firstHeu, secHeu);
+            topC.run();
+            auto topCNodes = topC.topkNodesList();
+            TopCloseness topCR(G, k, firstHeu, secHeu, nodeList);
+            topCR.run();
+            auto topCRNodes = topCR.topkNodesList();
+            auto topCRScores = topCR.topkScoresList();
+            EXPECT_EQ(topCNodes[0], topCRNodes[0]);
+            EXPECT_THAT(topCRNodes, testing::IsSupersetOf({0, 1, 2, 3, 4, 5, 26, 48, 64}));
+            EXPECT_THAT(topCNodes, testing::IsSupersetOf({26, 48, 64}));
+            EXPECT_THAT(topCNodes, testing::Not(testing::IsSupersetOf({0, 1, 2, 3, 4, 5})));
+            for (count i = 0; i < k - 1; i++) {
+                EXPECT_GE(topCRScores[i], topCRScores[i + 1]);
+            }
```

Comment:
I recommend to use `std::is_sorted` https://en.cppreference.com/w/cpp/algorithm/is_sorted